### PR TITLE
gdbserial/gdbserver: optimize the search of debugserver binary

### PR DESCRIPTION
**Pull request by chainhelen**
Unexport `GetDebugServerAbsolutePath` and avoid unnecessary repeated calls.
Remove `os.Stat` because `Exec.LookPath` has already used `os.Stat`.And Fix
some comments.

### CODE REVIEW

**Comment by chainhelen** [comment]
To be on the safe side, can you help me re-run ci for travis? thx.

**Comment by aarzilli** [comment]
Done.

**Comment by chainhelen** [comment]
Ha ha, so fortunate again.

**Comment by derekparker** [comment]
Hm, yeah in theory this should work. The only difference is before we always returned a path (the last fallback path was never looked up via ExecPath).

**Comment by chainhelen** [comment]
Ok, I want to try to test failed CI for this pr. So don't merge temporarily.

File: pkg/proc/gdbserial/gdbserver.go [hunk 1]
```diff
@@ -363,8 +361,8 @@ func LLDBLaunch(cmd []string, wd string, foreground bool, debugInfoDirs []string
 	var listener net.Listener
 	var port string
 	var process *exec.Cmd
-	if _, err := os.Stat(GetDebugServerAbsolutePath()); err == nil {
-		listener, err = net.Listen("tcp", "127.0.0.1:0")
+	if debugserverExecutable := getDebugServerAbsolutePath(); debugserverExecutable != "" {
+		listener, err := net.Listen("tcp", "127.0.0.1:0")
```
Comment (derekparker) [inline pkg/proc/gdbserial/gdbserver.go:365]: This is the cause of the breakage. This variable shadows the `listener` variable above, causing it to never be set, thus we end up with a zero-value listener even though we were able to find the debug server executable.

Comment (chainhelen) [reply pkg/proc/gdbserial/gdbserver.go]: Thx, done

**Pull request merged**